lib/CodeGen/CGExpr.cpp

```diff
@@ -2087,6 +2121,15 @@ ExprValuePtr CodeGenFunction::emitAsmSpecialCallExpr(const AsmIdentifier *SI,
   case AsmIdentifier::SpecialIdentifier::codecopy:
     emitAsmCallCodeCopy(CE);
     return std::make_shared<ExprValue>();
+  case AsmIdentifier::SpecialIdentifier::codesize:
+    return ExprValue::getRValue(
+        CE, Builder.CreateZExtOrTrunc(CGM.emitGetCodeSize(), CGM.Int256Ty));
+  case AsmIdentifier::SpecialIdentifier::extcodesize:
+    return ExprValue::getRValue(CE, emitAsmExternalGetCodeSize(CE));
```

Comment:
Some of the `CodeSize` need to do zero extensions or truncations. But some of them are not.
Maybe we should check all of the `CodeSize` to make sure the behaviors are all the same.

Reply:
Yes, I should make sure all the return type is int256. This is corrected in the new commit.